doc/meson.build

```diff
@@ -0,0 +1,26 @@
+if pandoc.found()
+    custom_target(
+        input: ['manpage-header.md', 'netplan.md', 'manpage-footer.md'],
+        output: 'netplan.5',
+        command: [pandoc, '-s', '-o', '@OUTPUT@', '@INPUT@'],
+        install: true,
+        install_dir: join_paths(get_option('mandir'), 'man5'))
+    custom_target(
+        input: 'netplan.md',
+        output: 'netplan.html',
+        command: [pandoc, '-s', '--metadata', 'title="Netplan reference"', '--toc', '-o', '@OUTPUT@', '@INPUT@'],
+        install: true,
+        install_dir: join_paths(get_option('datadir'), 'doc', 'netplan'))
+    foreach doc : ['netplan-apply', 'netplan-dbus', 'netplan-generate', 'netplan-get', 'netplan-set', 'netplan-try']
+        markdown = files(doc + '.md')
+        manpage = doc + '.8'
+        custom_target(
+            input: markdown,
+            output: manpage,
+            command: [pandoc, '-s', '-o', '@OUTPUT@', '@INPUT@'],
+            install: true,
+            install_dir: join_paths(get_option('mandir'), 'man8'))
+    endforeach
+else
+    warning('Program "pandoc" not found!  Cannot generate documentation/man pages')
+endif
```

Comment:
... and then you can skip this warning, because people who absolutely want the docs can configure it with enabled, people who are okay gracefully degrading can configure it with auto, and people who absolutely don't want a pandoc dependency can disable it and not get a big yellow warning.